packages/core/src/browser/shell/application-shell.ts

```diff
@@ -811,6 +811,45 @@ export class ApplicationShell extends Widget {
         return result;
     }
 
+    /**
+     * Finds the title widget from the tab-bar.
+     * @param tabBar used for providing an array of titles.
+     * @returns the selected title widget, else returns the currentTitle or undefined.
+     */
+    findTitle(tabBar: TabBar<Widget>, event?: Event): Title<Widget> | undefined {
+        if (event && event.target) {
+            let tabNode: HTMLElement | null = event.target as HTMLElement;
+            while (tabNode && !tabNode.classList.contains('p-TabBar-tab')) {
```

Comment:
[minor]: Can we use [`closest`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)? But I am fine with the current logic too.